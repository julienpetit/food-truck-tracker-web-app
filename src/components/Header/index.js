import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Menu } from 'semantic-ui-react';
import './Header.css';

const Header = ({ isAuthenticated = false }) => (
  <header className="App-header">
    <h1 className="App-title">Welcome to React</h1>
    <Menu pointing>
      <Menu.Item
        as={NavLink}
        to="/"
        exact
        activeClassName="active"
        name="Home"
      />
      <Menu.Item
        as={NavLink}
        to="/account"
        exact
        activeClassName="active"
        name="Account"
      />

      <Menu.Item
        as={NavLink}
        to="/register"
        exact
        activeClassName="active"
        name="Register"
      />

      <Menu.Item
        as={NavLink}
        to="/users"
        exact
        activeClassName="active"
        name="Users"
      />

      <Menu.Menu position="right">
        {isAuthenticated ? (
          <Menu.Item
            as={NavLink}
            to="/logout"
            exact
            activeClassName="active"
            name="Logout"
          />
        ) : (
          <Menu.Item
            as={NavLink}
            to="/login"
            exact
            activeClassName="active"
            name="Login"
          />
        )}
      </Menu.Menu>
    </Menu>
  </header>
);

Header.propTypes = {
  isAuthenticated: PropTypes.bool
};

export default Header;
