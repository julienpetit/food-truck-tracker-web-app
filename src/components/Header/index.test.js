import React from 'react';
import Header from './index';
import { shallowWithIntl } from '../../helpers/intl-enzyme-test-helper.js';

describe('always render subcomponents', () => {
  const wrapper = shallowWithIntl(<Header isAuthenticated={true} />).dive();

  test('always renders header', () => {
    const header = wrapper.find('header');
    expect(header.length).toBe(1);
  });

  test('always renders an h1', () => {
    const h1 = wrapper.find('h1');
    expect(h1.length).toBe(1);
  });
});

test('NavLink change when isAuthenticated is true or false', () => {
  let wrapper;

  wrapper = shallowWithIntl(<Header isAuthenticated={true} />);

  expect(wrapper).toMatchSnapshot();

  wrapper = shallowWithIntl(<Header isAuthenticated={false} />);

  expect(wrapper).toMatchSnapshot();
});
