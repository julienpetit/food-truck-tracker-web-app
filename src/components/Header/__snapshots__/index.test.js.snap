// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink change when isAuthenticated is true or false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isAuthenticated={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="App-title"
>
          Welcome to React
</h1>,
        <Menu
          pointing={true}
>
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Home"
                    to="/"
          />
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Account"
                    to="/account"
          />
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Register"
                    to="/register"
          />
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Users"
                    to="/users"
          />
          <MenuMenu
                    position="right"
          >
                    <MenuItem
                              activeClassName="active"
                              as={[Function]}
                              exact={true}
                              name="Logout"
                              to="/logout"
                    />
          </MenuMenu>
</Menu>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to React",
          "className": "App-title",
        },
        "ref": null,
        "rendered": "Welcome to React",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Home"
              to="/"
/>,
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Account"
              to="/account"
/>,
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Register"
              to="/register"
/>,
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Users"
              to="/users"
/>,
            <MenuMenu
              position="right"
>
              <MenuItem
                            activeClassName="active"
                            as={[Function]}
                            exact={true}
                            name="Logout"
                            to="/logout"
              />
</MenuMenu>,
          ],
          "pointing": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Account",
              "to": "/account",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Register",
              "to": "/register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Users",
              "to": "/users",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Logout"
                to="/logout"
/>,
              "position": "right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Logout",
                "to": "/logout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="App-title"
>
            Welcome to React
</h1>,
          <Menu
            pointing={true}
>
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Home"
                        to="/"
            />
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Account"
                        to="/account"
            />
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Register"
                        to="/register"
            />
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Users"
                        to="/users"
            />
            <MenuMenu
                        position="right"
            >
                        <MenuItem
                                    activeClassName="active"
                                    as={[Function]}
                                    exact={true}
                                    name="Logout"
                                    to="/logout"
                        />
            </MenuMenu>
</Menu>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Welcome to React",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Home"
                to="/"
/>,
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Account"
                to="/account"
/>,
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Register"
                to="/register"
/>,
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Users"
                to="/users"
/>,
              <MenuMenu
                position="right"
>
                <MenuItem
                                activeClassName="active"
                                as={[Function]}
                                exact={true}
                                name="Logout"
                                to="/logout"
                />
</MenuMenu>,
            ],
            "pointing": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Home",
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Account",
                "to": "/account",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Register",
                "to": "/register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Users",
                "to": "/users",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <MenuItem
                  activeClassName="active"
                  as={[Function]}
                  exact={true}
                  name="Logout"
                  to="/logout"
/>,
                "position": "right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClassName": "active",
                  "as": [Function],
                  "exact": true,
                  "name": "Logout",
                  "to": "/logout",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NavLink change when isAuthenticated is true or false 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isAuthenticated={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="App-title"
>
          Welcome to React
</h1>,
        <Menu
          pointing={true}
>
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Home"
                    to="/"
          />
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Account"
                    to="/account"
          />
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Register"
                    to="/register"
          />
          <MenuItem
                    activeClassName="active"
                    as={[Function]}
                    exact={true}
                    name="Users"
                    to="/users"
          />
          <MenuMenu
                    position="right"
          >
                    <MenuItem
                              activeClassName="active"
                              as={[Function]}
                              exact={true}
                              name="Login"
                              to="/login"
                    />
          </MenuMenu>
</Menu>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to React",
          "className": "App-title",
        },
        "ref": null,
        "rendered": "Welcome to React",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Home"
              to="/"
/>,
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Account"
              to="/account"
/>,
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Register"
              to="/register"
/>,
            <MenuItem
              activeClassName="active"
              as={[Function]}
              exact={true}
              name="Users"
              to="/users"
/>,
            <MenuMenu
              position="right"
>
              <MenuItem
                            activeClassName="active"
                            as={[Function]}
                            exact={true}
                            name="Login"
                            to="/login"
              />
</MenuMenu>,
          ],
          "pointing": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Account",
              "to": "/account",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Register",
              "to": "/register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "as": [Function],
              "exact": true,
              "name": "Users",
              "to": "/users",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Login"
                to="/login"
/>,
              "position": "right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Login",
                "to": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="App-title"
>
            Welcome to React
</h1>,
          <Menu
            pointing={true}
>
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Home"
                        to="/"
            />
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Account"
                        to="/account"
            />
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Register"
                        to="/register"
            />
            <MenuItem
                        activeClassName="active"
                        as={[Function]}
                        exact={true}
                        name="Users"
                        to="/users"
            />
            <MenuMenu
                        position="right"
            >
                        <MenuItem
                                    activeClassName="active"
                                    as={[Function]}
                                    exact={true}
                                    name="Login"
                                    to="/login"
                        />
            </MenuMenu>
</Menu>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Welcome to React",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Home"
                to="/"
/>,
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Account"
                to="/account"
/>,
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Register"
                to="/register"
/>,
              <MenuItem
                activeClassName="active"
                as={[Function]}
                exact={true}
                name="Users"
                to="/users"
/>,
              <MenuMenu
                position="right"
>
                <MenuItem
                                activeClassName="active"
                                as={[Function]}
                                exact={true}
                                name="Login"
                                to="/login"
                />
</MenuMenu>,
            ],
            "pointing": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Home",
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Account",
                "to": "/account",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Register",
                "to": "/register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "as": [Function],
                "exact": true,
                "name": "Users",
                "to": "/users",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <MenuItem
                  activeClassName="active"
                  as={[Function]}
                  exact={true}
                  name="Login"
                  to="/login"
/>,
                "position": "right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClassName": "active",
                  "as": [Function],
                  "exact": true,
                  "name": "Login",
                  "to": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
